ethereum/dex/liquidity/insert_balancer_v2.sql

```diff
@@ -0,0 +1,156 @@
+CREATE OR REPLACE FUNCTION dex.insert_liquidity_balancer_v2(start_ts timestamptz, end_ts timestamptz=now()) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+WITH days as ( -- update table entries until previous day
+    SELECT day FROM generate_series(start_ts, (SELECT end_ts - interval '1 day'), '1 day') g(day)
+),
+balancer_v2_pools as ( -- https://github.com/duneanalytics/abstractions/blob/master/labels/ethereum/balancer_v2_pools.sql
+    SELECT
+        pool_id,
+        SUBSTRING(pool_id FOR 20) as pool_address,
+        token_address,
+        normalized_weight,
+        symbol as pool_name_symbolic
+    FROM
+    (
+    select c."poolId" as pool_id, unnest(cc.tokens) as token_address, unnest(cc.weights)/1e18 as normalized_weight, cc.symbol, 'WP' as pool_type
+    from balancer_v2."Vault_evt_PoolRegistered" c
+    inner join balancer_v2."WeightedPoolFactory_call_create" cc
+    on c.evt_tx_hash = cc.call_tx_hash
+    union all
+    select c."poolId" as pool_id, unnest(cc.tokens) as token_address, unnest(cc.weights)/1e18 as normalized_weight, cc.symbol, 'WP2T' as pool_type
+    from balancer_v2."Vault_evt_PoolRegistered" c
+    inner join balancer_v2."WeightedPool2TokensFactory_call_create" cc
+    on c.evt_tx_hash = cc.call_tx_hash
+    union all
+    select c."poolId" as pool_id, unnest(cc.tokens) as token_address, 0 as normalized_weight, cc.symbol, 'SP' as pool_type
+    from balancer_v2."Vault_evt_PoolRegistered" c
+    inner join balancer_v2."StablePoolFactory_call_create" cc
+    on c.evt_tx_hash = cc.call_tx_hash
+    ) all_pools
+),
+balance_changes AS (
+SELECT change.evt_block_time AS timestamp
```

Comment:
This looks like a typo, remove it or add the comma at the end of the line.

```suggestion
SELECT
```

Reply:
hey again @mendesfabio, 
sorry it took me a bit longer to get this finished.

`dex.liquidity` is now available in Dune's Ethereum database. You can use and query this table but please proceed with caution. The entries for some pools shows erroneous negative token balances (~ 0.7%). These come from the underlying erc20.token_balances table and have been reported to the upstream engineers. This issue also affects Balancer v1.

Please see this query for Balancer v2: https://dune.xyz/queries/132478. Might be good to know that I did not yet implement the logic for the column `token_index` here. Some DEXes have a specific order for the pool tokens.

Hope this helps! The credit for the Balancer code goes to you and @markusbkoch. I just tried to integrate your code into a more general framework. Also trying to keep things simple here by relying on existing abstractions such as `prices.usd`. All PRs welcome :)